include/dem/checkpointing.h

```diff
@@ -0,0 +1,96 @@
+/* ---------------------------------------------------------------------
+ *
+ * Copyright (C) 2019 - 2019 by the Lethe authors
+ *
+ * This file is part of the Lethe library
+ *
+ * The Lethe library is free software; you can use it, redistribute
+ * it, and/or modify it under the terms of the GNU Lesser General
+ * Public License as published by the Free Software Foundation; either
+ * version 2.1 of the License, or (at your option) any later version.
+ * The full text of the license can be found in the file LICENSE at
+ * the top level of the Lethe distribution.
+ *
+ * ---------------------------------------------------------------------
+
+ *
+ * Author: Shahab Golshan, Polytechnique Montreal, 2019
+ */
+
+#include <deal.II/base/timer.h>
+
+#include <deal.II/distributed/tria.h>
+
+#include <deal.II/particles/particle_handler.h>
+
+#include <boost/archive/text_iarchive.hpp>
+#include <boost/archive/text_oarchive.hpp>
+#include <iostream>
+#include <fstream>
+#include <core/pvd_handler.h>
+#include <dem/dem_solver_parameters.h>
+
+
+using namespace dealii;
+
+#ifndef checkpointing_h
+#  define checkpointing_h
+
+/**
+ * Carries out reading and writing dem checkpoints
+ *
+ * @note
+ *
+ * @author Shahab Golshan, Polytechnique Montreal 2019-
+ */
+
+template <int dim>
+class Checkpointing
```

Comment:
Clean stuff. Maybe class should be called DEMCheckpointing?

Reply:
Actually, why does this need to be a class? Could it not be just two functions? Class logic does not appear to be used here.

Reply:
done